hubcap/version.py

```diff
@@ -12,7 +12,7 @@
 def parse_semver_tag(tag):
     '''use regexes to parse the semver tag into groups'''
     # regex taken from official SEMVER documentation site
```

Comment:
Does this regex now slightly deviate from the official site? Or had it moved forward and we hadn't moved with it?

Reply:
The regex in the `main` branch currently matches the "named groups" regular expression in [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) precisely:
`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`

That regex has not changed (and probably will not change within SemVer 2.x).

This PR just adds a single character change that allows for the dash prefix for pre-releases to be optional:
<img width="687" alt="Screen Shot 2022-07-26 at 9 08 31 AM" src="https://user-images.githubusercontent.com/44704949/181043388-6c3f4ae7-e373-4b04-8bb9-3973c81e14d5.png">

This would to allow for dbt package versions to have greater overlap with [PEP 440](https://peps.python.org/pep-0440/) even though SemVer 2.0.0 doesn't allow it to be optional.

Reply:
Yeah so my point is that if it's no longer the official SEMVER regex, we should update the comment to say something like "based on the SEMVER regex but with tweaks for PEP compatibility"